File: app/src/main/java/com/example/toothfairy/data/WearingTime.java
package com.example.toothfairy.data;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@ToString
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class WearingTime {
    Long start; // milliseconds
    Long end;   // milliseconds

    public Long getWearingTime(){
        return end - start;
    }
    public int getHour(){
        return (int)((end - start) / (1000 * 60)) / 60;
    }
    public int getMinute(){
        return (int)((end - start) / (1000 * 60)) % 60;
    }
}

File: app/src/main/java/com/example/toothfairy/data/WearingTime.kt
package com.example.toothfairy.data

class WearingTime {
    var start: Long? = null  // milliseconds
    var end: Long? = null // milliseconds

    constructor(start: Long?, end: Long?) {
        this.start = start
        this.end = end
    }

    constructor() {}

    val wearingTime: Long
        get() = end!! - start!!
    val hour: Int
        get() = ((end!! - start!!) / (1000 * 60)).toInt() / 60
    val minute: Int
        get() = ((end!! - start!!) / (1000 * 60)).toInt() % 60

    override fun toString(): String {
        return "WearingTime(start=$start, end=$end)"
    }
}